### ISSUE TITLE
New WooCommerce Navigation Compatibility With PublishPress

### ISSUE THREAD

**Issue by theabhig** [woocommerce/woocommerce-admin#6078]
**Describe the bug**

**_I'm fairly confident this is out of scope and up to the third party plugin developer to look into but just in case it is a different bug I wanted to log it here_**

A user in [live chat](https://mc.a8c.com/support-stats/happychat/transcript.php?id=26599198) is using PublishPress to manage their WooCommerce store permissions.

They're on an eCommerce plan and had the [new navigation](https://wordpress.com/support/new-navigation-in-the-ecommerce-plan/) rolled out today.

And their store access permissions stopped working. Users with the custom 'Store owner' can only see the existing orders.

**To Reproduce**

1. Go to https://elbasam.com/wp-admin/edit.php?post_type=shop_order
2. From here, a user with the store owner permissions should be able to see products, orders and most of the admin permissions.
3. Users can only see the information here: https://i.snipboard.io/FJ4Xui.jpg
4. Permission levels are set here using the plugin: https://elbasam.com/wp-admin/admin.php?page=capsman

**Expected behavior**

Users with 'Store owner' permissions should be able to see the other areas of WooCommerce in their sidebar.

**Screenshots**

https://i.snipboard.io/FJ4Xui.jpg

**Desktop (please complete the following information):**

Screen Size: 1366 x 768
Browser Size: 1366 x 625
User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.141 Safari/537.36
Local Date: ١١:٤٨ ص، يناير ١٥ ٢٠٢١
Location: Antioch, Turkey

**Additional context**

There was a language barrier in live chat so some of the information might not be entirely clear or slightly lost in translation. 

Also even when the site was switched to English (from Arabic), some of the settings and options remained in Arabic making it a bit more difficult to troubleshoot.

User: https://mc.a8c.com/tools/reportcard/user/?id=185522033

Store admin: https://wordpress.com/wp-admin/network/wpcom-paid-upgrades.php?action=search&username=mrbalshah

**Comment by joshuatf** [comment]
Thanks for opening the issue @theabhig!

I'll have more time to dive into this next week, but it sounds like this is related to how permissions are registered in PublishPress or potentially how they are set up in the new navigation.

This will require a bit more investigation, but probably means a PR needs to be opened in PublishPress to address.

**Comment by joshuatf** [comment]
@theabhig As an update to this, I was able to install PublishPress and create a custom role that was not able to view specific navigation items.

After digging into this a bit, I noticed a few bugs with PublishPress, specifically around how capabilities are registered with a user.  Dumping the information from:
```php
$user_data = get_userdata( get_current_user_id() );
$caps = $user_data->allcaps;
```

I was able to see that not all checked capabilities had made it it into the roles.  Typically the the two most generic capabilities needed for these screens are `manage_woocommerce` and `view_woocommerce_reports`.  Checking both of those gives me access to those screens.  `edit_shop_orders` grants access to the order items.  

I tried adding these capabilities via another plugin ( https://wordpress.org/plugins/leira-roles/ ) and was able to successfully view those menu items and respective screens.

I did notice a couple oddities around categories shown when no child items existed and the fact the we need `view_woocommerce_reports` in order to access the navigation for any WC pages now, but I believe those are unrelated to the original issue and I'll open separate issues to address those.

**Comment by psealock** [comment]
Maybe this will be fixed by https://github.com/woocommerce/woocommerce-admin/pull/6366 and/or https://github.com/woocommerce/woocommerce-admin/pull/6365 which are set to be included in a 2.1.0 release coming up next week and included in the eComm plan in March's WC release.

**Comment by joelclimbsthings** [comment]
Sounds like this was likely addressed by https://github.com/woocommerce/woocommerce-admin/pull/6366 and/or https://github.com/woocommerce/woocommerce-admin/pull/6365 as mentioned, closing in preparation for the merge.
